src/doom/hu_stuff.c

```diff
@@ -217,11 +217,7 @@ char*   mapnames_chex[] =   // Chex Quest names.
     "NEWLEVEL"
 };
 
-// List of names for levels in commercial IWADs
```

Comment:
I took that into account when looking into this issue. Storing all three IWAD names in a single array is part of the problem: Final Doom appears to have stored the Plutonia and TNT mapnames contiguously, but Doom II didn't know about either of these, which is why the original report had this complaint:

> 1- The name of the level in the automap (with DOOM2.EXE it's ?* !? /# 8/ ?$@ /* or something like that and Chocolate-Doom.exe is Level 1: Congo)

I tested vanilla Doom II's situation in DOSBox and found that the automap names were blank, with the same being true of vanilla TNT: by comparison, Chocolate Doom II spills into Plutonia's names and Chocolate TNT map33 read "M_SKULL1". While there is overrun behavior going on, the only coherent case appears to apply to Plutonia.

I'll test my changes against PL2.WAD specifically since that seems to be the worrisome case. As I recall, D2TWID.WAD also stores a map33, so I'll check its compatibility as well.